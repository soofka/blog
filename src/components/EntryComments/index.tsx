import * as React from 'react';

interface EntryCommentsPropsInterface {
  title: string;
  url: string;
  shortName: string;
  identifier: string;
}

declare const DISQUS: any;

export class EntryComments extends React.Component<EntryCommentsPropsInterface, {}> {

  private disqusSetupAttempts: number = 0;

  componentDidMount(): void {
    const { title, url, shortName, identifier } = this.props;

    const disqusSetupAttempt = setInterval(
      () => {
        if (typeof DISQUS !== 'undefined') {
          DISQUS.reset({
            reload: true,
            config() {
              this.page.title = title;
              this.page.url = window.location.href;
              this.page.shortName = shortName;
              this.page.identifier = identifier;
            },
          });
          clearInterval(disqusSetupAttempt);
        }

        this.disqusSetupAttempts += 1;
        if (this.disqusSetupAttempts >= 3) {
          clearInterval(disqusSetupAttempt);
        }
      },
      1000,
    );
  }

  render(): JSX.Element {
    return (
      <div id="disqus_thread"></div>
    );
  }
}

export default EntryComments;
