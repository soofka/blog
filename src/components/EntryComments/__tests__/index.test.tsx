import * as React from 'react';
import { shallow } from 'enzyme';

import EntryComments from 'components/EntryComments';

describe('<EntryComments/>', () => {

  it('renders correctly', () => {
    expect(shallow(
      <EntryComments
        title="testTitle"
        url="testUrl"
        shortName="testShortName"
        identifier="testIdentifier"
      />)).toMatchSnapshot();
  });

});
