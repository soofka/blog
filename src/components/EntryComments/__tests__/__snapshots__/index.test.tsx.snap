// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EntryComments/> renders correctly 1`] = `
<div
  id="disqus_thread"
/>
`;
